### ISSUE TITLE
Some questions about gradient backpropagation

### ISSUE THREAD

**Issue by Kai-Sheng-caesar** [LINs-lab/DynMoE#6]
I've recently been experimenting with some MoE architectures. However, due to non-differentiable operations like the top-k selection or other non-differentiable components, gradients cannot backpropagate properly. So the learnable parameters I added before the top-k operation failed to update during training. I’d like to ask how did you implement the learnable probability thresholds for each expert network? Thank you!
